cmd/spire-server/cli/run/run.go

```diff
@@ -414,6 +415,27 @@ func NewServerConfig(c *Config, logOptions []log.Option, allowUnknownConfig bool
 				},
 			}
 
+			if c.Server.Federation.BundleEndpoint.RefreshHint != "" {
+				refreshHint, err := time.ParseDuration(c.Server.Federation.BundleEndpoint.RefreshHint)
+				if err != nil {
+					return nil, fmt.Errorf("Could not parse refresh_hint %q: %w", c.Server.Federation.BundleEndpoint.RefreshHint, err)
+				}
+
+				if refreshHint >= 24*time.Hour {
+					sc.Log.Warn("Bundle endpoint refresh hint set to a high value. " +
+						"It's recommended to set the refresh hint to a small value" +
+						", e.g. 5 minutes to make sure that trust domains that federate " +
+						"with this trust domain refresh the trust bundle often enough to " +
+						"cover the case of unscheduled trust bundle updates.")
```

Comment:
nit: The message could probably be more concise, something like:
"Bundle endpoint refresh hint set to a high value. To cover the case of unscheduled trust bundle updates, it's recommended to have a smaller value, e.g. 5m"